packages/mantine/src/components/button/Button.tsx

```diff
@@ -41,7 +44,7 @@ export const Button = polymorphicFactory<ButtonOverloadFactory>(
                     loaderProps={{variant: 'oval'}}
                     ref={ref}
                     loading={isLoading || loading}
-                    onClick={handleClick}
+                    onClick={dataDisabled ? (e) => e.preventDefault() : handleClick}
```

Comment:
I think we should continue to forward the value to Mantine to apply the style:
```suggestion
                    onClick={dataDisabled ? (e) => e.preventDefault() : handleClick}
                    data-disabled={dataDisabled}
```

Reply:
`Line 48` fixed the visual style of the disabled button :)